Cut(eos_designs)!: Remove deprecated keys for underlay multicast (#5860)
aristanetworks/ansible-avd: docs/porting-guides/6.x.x.md

| `source_interfaces.radius` | `aaa_settings.radius.vrfs[].source_interface or aaa_settings.radius.servers[].vrf: use_mgmt_interface_vrf or aaa_settings.radius.servers[].vrf: use_inband_mgmt_vrf` |
| `source_interfaces.snmp` | `snmp_settings.vrfs[].source_interface or snmp_settings.hosts[].vrf: use_mgmt_interface_vrf or snmp_settings.hosts[].vrf: use_inband_mgmt_vrf or avd_6_behaviors.snmp_settings_vrfs` |
| `source_interfaces.tacacs` | `aaa_settings.tacacs.vrfs[].source_interface or aaa_settings.tacacs.servers[].vrf: use_mgmt_interface_vrf or aaa_settings.tacacs.servers[].vrf: use_inband_mgmt_vrf` |
| `underlay_mutlticast` | `underlay_mutlticast_pim_sm` or `<node_type_key>.defaults.underlay_multicast.pim_sm.enabled` or `<node_type_key>.node_groups[].underlay_multicast.pim_sm.enabled` or `<node_type_key>.node_groups[].nodes[].underlay_multicast.pim_sm.enabled` or `<node_type_key>.nodes[].underlay_multicast.pim_sm.enabled` |
| `underlay_ospf_authentication.message_digest_keys[].key` | `underlay_ospf_authentication.message_digest_keys[].cleartext_key` |

### Change of default behavior for PIM-SM multicast for `core_interfaces.p2p_links` and `l3_edge.p2p_links`

> In AVD 5.x, to enable PIM SM for a `core_interface` or `l3_edge` link, it was required to have the global `underlay_mutlicast: true` and under the P2P link (or its profile)
> `include_in_underlay_protocol: true` and `underlay_mutlicast: true`. If any condition was not set, PIM SM was not enabled.

Reviewer: ```suggestion
In AVD 5.X.X, enabling PIM-SM on a `core_interface` or `l3_edge` link required all of the following settings:
  - `underlay_multicast: true` at the global level
  - `underlay_multicast: true` **and** `include_in_underlay_protocol: true` at the link or link profile level
```

| `core_interfaces.p2p_links_profiles[].underlay_multicast` | `core_interfaces.p2p_links_profiles.multicast_pim_sm` |
| `l3_edge.p2p_links[].underlay_multicast` | `l3_edge.p2p_links.multicast_pim_sm` |
| `l3_edge.p2p_links_profiles[].underlay_multicast` | `l3_edge.p2p_links_profiles.multicast_pim_sm` |
| `name_servers` | `dns_settings.servers` |
| `node_type_keys[].interface_descriptions.overlay_loopback_interface` | `node_type_keys[].interface_descriptions.router_id_loopback_interface`|
| `source_interfaces.domain_lookup` | `dns_settings` |
| `source_interfaces.radius` | `aaa_settings.radius.vrfs[].source_interface or aaa_settings.radius.servers[].vrf: use_mgmt_interface_vrf or aaa_settings.radius.servers[].vrf: use_inband_mgmt_vrf` |
| `source_interfaces.snmp` | `snmp_settings.vrfs[].source_interface or snmp_settings.hosts[].vrf: use_mgmt_interface_vrf or snmp_settings.hosts[].vrf: use_inband_mgmt_vrf or avd_6_behaviors.snmp_settings_vrfs` |
| `source_interfaces.tacacs` | `aaa_settings.tacacs.vrfs[].source_interface or aaa_settings.tacacs.servers[].vrf: use_mgmt_interface_vrf or aaa_settings.tacacs.servers[].vrf: use_inband_mgmt_vrf` |

> | `underlay_mutlticast` | `underlay_mutlticast_pim_sm` or `<node_type_key>.defaults.underlay_multicast.pim_sm.enabled` or `<node_type_key>.node_groups[].underlay_multicast.pim_sm.enabled` or `<node_type_key>.node_groups[].nodes[].underlay_multicast.pim_sm.enabled` or `<node_type_key>.node_groups[].underlay_multicast.pim_sm.enabled`or `<node_type_key>.node_groups[].underlay_multicast.pim_sm.enabled` or `<node_type_key>.nodes[].underlay_multicast.pim_sm.enabled` |

Reviewer: Still early in the morning for me but something feels wrong here. I think there are duplicates.

Author: no you are right :p

Author: ```suggestion
| `underlay_mutlticast` | `underlay_mutlticast_pim_sm` or `<node_type_key>.defaults.underlay_multicast.pim_sm.enabled` or `<node_type_key>.node_groups[].underlay_multicast.pim_sm.enabled` or `<node_type_key>.node_groups[].nodes[].underlay_multicast.pim_sm.enabled` or `<node_type_key>.nodes[].underlay_multicast.pim_sm.enabled` |
```

> | `underlay_ospf_authentication.message_digest_keys[].key` | `underlay_ospf_authentication.message_digest_keys[].cleartext_key` |

Reviewer: This is not part of this PR, is it?

Author: it was in the wrong place from an alphabetical point of view so I moved it a round - not sure where it s now

Author: cf removed line 60

> ### Change of default behavior for PIM SM multicast for `core_interfaces.p2p_links` and `l3_edge.p2p_links`

Reviewer: ```suggestion
### Change of default behavior for PIM-SM multicast for `core_interfaces.p2p_links` and `l3_edge.p2p_links`
```

In AVD 5.x, to enable PIM SM for a `core_interface` or `l3_edge` link, it was required to have the global `underlay_mutlicast: true` and under the P2P link (or its profile)

> `include_in_underlay_protocol: true` and `underlay_mutlicast: true`. If any condition was not set, PIM SM was not enabled.

Reviewer: I think it would be cleaner to have something like: "it was required to have the following knobs enabled: " and then have bullets points for each.

Author: please suggest :P

> As of AVD 6.0.0, the global `underlay_mutlicast` is removed and replaced with node setting `underlay_multicast.pim_sm.enabled` or global `underlay_mutlicast_pim_sm`.

Reviewer: ```suggestion
As of AVD 6.0.0, the global `underlay_mutlicast` is replaced with `underlay_mutlicast_pim_sm` or the node setting `underlay_multicast.pim_sm.enabled`.
```

Author: no we want to keep the precedence order

Author: ```suggestion
As of AVD 6.0.0, the global `underlay_mutlicast` is replaced with node setting `underlay_multicast.pim_sm.enabled` or global `underlay_mutlicast_pim_sm`.
```

Additionally `core_interfaces.p2p_links[].underlay_multicast` and `l3_edge.p2p_links[].underlay_multicast` are removed and replaced with respectively

> `core_interfaces.p2p_links[].multicast_pim_sm` and `l3_edge.p2p_links[].multicast_pim_sm`.

Reviewer: ```suggestion
Additionally, `core_interfaces.p2p_links[].underlay_multicast` and `l3_edge.p2p_links[].underlay_multicast` are replaced with
`core_interfaces.p2p_links[].multicast_pim_sm` and `l3_edge.p2p_links[].multicast_pim_sm` respectively.
```

Author: ```suggestion
Additionally, the `underlay_multicast` key has been replaced by `multicast_pim_sm` for both `core_interfaces.p2p_links[]` and `l3_edge.p2p_links[]`.
```

> The breaking change in default behavior is that for a P2P link, when `include_in_underlay_protocol: true`, then by default PIM SM is configured if enabled globally (whether by the node_settings or the global key).

Reviewer: ```suggestion
The breaking change in default behavior is for `l3_edge` and `core_interfaces` P2P links. PIM-SM is now automatically configured on links when `include_in_underlay_protocol: true`, provided that PIM-SM is already enabled globally (either through node settings or the global key).
```

Reviewer: Actually we should probably say "The breaking change in default behavior is for `l3_edge` and `core_interfaces` links".

Author: P2P links is how they are named under each model

To retain the previous behavior, the following change is required:

```diff
- underlay_multicast: true
# # The next line enables globally
+ underlay_mutlicast_pim_sm: true

 l3leafs:
   [...]
   nodes:
     - name: MULTICAST_LEAF

> +      # Alternatively, PIM SM can be enabled at node_setting level

Reviewer: ```suggestion
+      # Alternatively, PIM-SM can be enabled at the node setting level
```

+      underlay_multicast:
+        pim_sm:
+          enabled: true

 # The same applies for l3_edge
 core_interfaces:
   p2p_links:
     - nodes: [MULTICAST_LEAF, MULTICAST_PEER]
       [...]
       include_in_underlay_protocol: true
+      # previously `underlay_multicast` had a default of `false`
+      multicast_pim_sm: false
```

> !!! note

Reviewer: ```suggestion
!!! tip
```
